app/models/grade.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class Grade < ApplicationRecord
+  belongs_to :project
+  belongs_to :grader, class_name: "User", foreign_key: :user_id
+  belongs_to :assignment
+
+  validates :grade, :user_id, :project_id, :assignment_id, presence: true
+  validate :grading_scale, :assignment_project
+
+  private
+    def grading_scale
+      valid = case assignment&.grading_scale
```

Comment:
We are validating assignment id, ie assignment record must be always present, so no need of `&` here